sympy/simplify/tests/test_simplify.py

```diff
@@ -938,6 +938,11 @@ def test_simplify_kroneckerdelta():
                                      [0, K(0, n), 0, K(1, n)],
                                      [0, 0, K(0, n), 0],
                                      [0, 0, 0, K(0, n)]])
+    assert simplify(eye(1) * KroneckerDelta(0, n) *
+                    KroneckerDelta(1, n)) == Matrix([[0]])
+
+    assert simplify(S.Infinity * KroneckerDelta(0, n) *
```

Comment:
@smichr Here is the actual test that will break if 0 * m is removed. I added the earlier just because I didn't want to remove it...